client/src/lib.rs

```diff
@@ -17,13 +17,26 @@ pub trait Tpm {
     fn transact(&mut self, command: &[u8], response: &mut [u8]) -> TpmResult<()>;
 }
 
+// These methods are syntatic sugar for calling run_command* with a specific command.
+
 pub fn get_capability<T: Tpm>(
     tpm: &mut T,
     command: &GetCapabilityCmd,
 ) -> TpmResult<GetCapabilityResp> {
     run_command(command, tpm)
 }
 
+pub fn create_primary<T: Tpm>(
+    tpm: &mut T,
+    command: &CreatePrimaryCmd,
+    primary_handle: TpmiRhHierarchy,
+) -> TpmResult<(CreatePrimaryResp, TPM2Handle)> {
+    let mut session = PasswordSession::default();
+    let mut sessions = CmdSessions::default();
```

Comment:
I am not 100% sure how sessions work in tpms yet, but this does not feel right to me, the three session handles are propagated down from user to ESAPI as here https://github.com/tpm2-software/tpm2-tss/blob/ba2ea649212388f6d928f6598a6f388023a77599/src/tss2-esys/api/Esys_CreatePrimary.c#L96C1-L97C22.

and I imagine it will be the same down to tpm ? am I missing something?

Reply:
Yeah so this is currently assuming you want a single password session for simplicity. When/if there's a use case for more sessions, other helpers can be added or more parameters can be plumbed through this one.

Reply:
I am not fully convinced what use case you have in mind, but aside from this small thing, I think that patch look fine to me,I built few things on top and testing.


@chrisfenner  will it be possible to approve this patch soon as I already have a patch that builds on top.

Reply:
Frankly, I agree that this seems incorrect. I'll make some suggestions in the near future for how to abstract this client.